pennylane/workflow/execution.py

```diff
@@ -0,0 +1,51 @@
+# Copyright 2018-2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Contains the :class:`ExecutionConfig` data class.
+"""
+from dataclasses import dataclass, field
+from typing import Callable, Union
+
+from pennylane.interfaces import SUPPORTED_INTERFACES
+
+
+@dataclass
+class ExecutionConfig:
+    """
+    A configuration class to describe an execution of a quantum circuit on a device.
+    """
+
+    shots: Union[None, int] = None
+    """The number of shots for an execution"""
+
+    gradient_method: Union[None, Callable] = None
+    """The method used to compute the gradient of the quantum circuit being executed"""
+
+    hyperparameters: dict = field(default_factory=dict)
```

Comment:
I'm a little unsure where these parameters would be coming from and where they'd be getting used. Can you give me an example?

Reply:
idk. the story said gradient hyperparameters should be configurable here, so I added them :p

Reply:
Ah. So what I'm meant is [these things](https://github.com/PennyLaneAI/pennylane/blob/359130c783b48fea04bac38c5602df2cc3aa401d/pennylane/gradients/gradient_transform.py#L22). In QNode, they are handled fairly suboptimally as the `gradient_kwargs`.

Maybe just `gradient_hyperparameters` instead? With the description that they are used in by the gradient transform.